script.sh

```
@@ -1,1726 +1,605 @@
+}
 
-  # Validate
-  if [[ -z "$DB_HOST" || -z "$DB_PORT" || -z "$DB_NAME" || -z "$DB_USER" || -z "$DB_PASSWORD" ]]; then
-    echo "Error: Database details are missing or invalid in config.yaml."
-    exit 1
+# Function to test PostgreSQL connection
+test_postgres_connection() {
+  print_color "yellow" "\nAttempting to connect to the database..."
+  
+  # Use .pgpass file for secure password handling
+  local pgpass_file=$(mktemp)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)